### PR TITLE
cdc add docs about handle-key only

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-sink-to-kafka.md [hunk 1]
```diff
@@ -287,3 +287,66 @@ write-key-threshold = 30000
...
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.TIKV_REGION_STATUS WHERE DB_NAME="database1" AND TABLE_NAME="table1" AND IS_INDEX=0;
 ```
+
+## 处理超过 Kafka Topic 限制的消息
+
+Kafka Topic 对可以接收的消息大小有限制，由 [max.message.bytes](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
+
+### 只发送 Handle-Key 部分数据
+
+从 v7.3.0 开始，TiCDC Kafka sink 支持只发送 Handle-Key 部分数据, 这样可以大大减少消息的大小，从而避免超过 Kafka Topic 限制。 Handle-Key 指的是：
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-kafka.md:297]: TiCDC Kafka sink 支持在消息超过限制的时候只发送 Handle-Key 部分数据

File: ticdc/ticdc-sink-to-kafka.md [hunk 2]
```diff
@@ -287,3 +287,66 @@ write-key-threshold = 30000
 ```sql
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.TIKV_REGION_STATUS WHERE DB_NAME="database1" AND TABLE_NAME="table1" AND IS_INDEX=0;
 ```
+
+## 处理超过 Kafka Topic 限制的消息
+
+Kafka Topic 对可以接收的消息大小有限制，由 [max.message.bytes](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
+
+### 只发送 Handle-Key 部分数据
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-kafka.md:295]: 只发送 Handle-key

Comment (3AceShowHand) [inline ticdc/ticdc-sink-to-kafka.md:295]:
```suggestion
### 只发送 Handle-Key
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 3]
```diff
@@ -287,3 +287,66 @@ write-key-threshold = 30000
...
+* 如果被同步的表有定义 Primary Key，即为 Primary Key。
+* 如果没有 Primary Key，但是有定义 Not NULL Unique Key，即为 Unique Key。
+
+目前，该功能至此 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，用户需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
+
+配置样例如下所示：
+
+```toml
+[sink.kafka-config.large-message-handle]
+# 默认为空，设置为 "handle-key-only" 时，以打开此功能。
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-kafka.md:308]: 默认为空，消息超过大小限制后，同步任务失败。
设置为“ handle-key-only”，如果消息超过大小，data 字段内容只发送 handle-key ，如果还超过大小，同步任务失败。

File: ticdc/ticdc-sink-to-kafka.md [hunk 4]
```diff
@@ -287,3 +287,66 @@ write-key-threshold = 30000
...
```
Comment (3AceShowHand) [inline ticdc/ticdc-sink-to-kafka.md:308]:
```suggestion
# 默认为空，消息超过大小限制后，同步任务失败。
# 设置为 "handle-key-only"，如果消息超过大小，data 字段内容只发送 handle-key ，如果依旧超过大小，同步任务失败。
```

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-sink-to-kafka.md [hunk 5]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
 ```sql
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.TIKV_REGION_STATUS WHERE DB_NAME="database1" AND TABLE_NAME="table1" AND IS_INDEX=0;
 ```
+
+## 处理超过 Kafka Topic 限制的消息
+
+Kafka Topic 对可以接收的消息大小有限制，由 [max.message.bytes](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:293]:
```suggestion
Kafka Topic 对可以接收的消息大小有限制，由 [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 6]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+### 只发送 Handle-Key
+
+从 v7.3.0 开始，TiCDC Kafka sink 支持在消息超过限制的时候只发送 Handle-Key 部分数据, 这样可以大大减少消息的大小，从而避免因为消息大小超过 Kafka Topic 限制，进而导致 changefeed 发生错误，同步任务失败的情况。 Handle-Key 指的是：
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:297]:
```suggestion
从 v7.3.0 开始，TiCDC Kafka sink 支持在消息超过限制的时候只发送 Handle-Key 部分数据。这样可以显著减少消息的大小，避免因为消息大小超过 Kafka Topic 限制而导致 changefeed 发生错误和同步任务失败的情况。Handle-Key 指的是：
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 7]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+* 如果被同步的表有定义 Primary Key，即为 Primary Key。
+* 如果没有 Primary Key，但是有定义 Not NULL Unique Key，即为 Unique Key。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:300]:
```suggestion
* 如果被同步的表有定义主键，即为主键。
* 如果没有主键，但是有定义 Not NULL Unique Key，即为 Unique Key。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 8]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+目前，该功能至此 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，用户需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:302]:
```suggestion
目前，该功能支持 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，你需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 9]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+配置样例如下所示：
+
+```toml
+[sink.kafka-config.large-message-handle]
+# 默认为空，消息超过大小限制后，同步任务失败。
+# 设置为 "handle-key-only"，如果消息超过大小，data 字段内容只发送 handle-key ，如果依旧超过大小，同步任务失败。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:309]:
```suggestion
# 设置为 "handle-key-only"，如果消息超过大小，data 字段内容只发送 handle-key，如果依旧超过大小，同步任务失败。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 10]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+large-message-handle-option = "handle-key-only"
+```
+
+#### 消费只有 Handle-Key 的消息
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:313]:
```suggestion
### 消费只有 Handle-Key 的消息
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 11]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+    ],
+    "old": null,
+    "_tidb": {     // TiDB 的扩展字段
+        "commitTs": 163963314122145239,
+        "onlyHandleKey": true
+    }
+}
+```
+
+Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。用户需要查选上游 TiDB，通过 [tidb_snapshot 读取历史数据](/read-historical-data.md) 读取完整的数据。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:349]:
```suggestion
Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。用户需要查选上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 12]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+> **警告：**
+>
+> Kafka 消费者在处理数据，查询 TiDB 时，可能发生数据已经被 GC 的情况。用户需要[调整 TiDB 集群的 GC Lifetime 设置](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 为一个较大的值，以避免该情况。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:353]:
```suggestion
> 在 Kafka 消费者处理数据并查询 TiDB 时，可能发生数据已经被 GC 的情况。你需要[调整 TiDB 集群的 GC Lifetime 设置](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 为一个较大的值，以避免该情况。
```

File: ticdc/ticdc-sink-to-kafka.md (hunk 11 above)
Comment (ran-huang) [reply ticdc/ticdc-sink-to-kafka.md:349]: “查选上游 TiDB”是什么意思？

Comment (ran-huang) [reply ticdc/ticdc-sink-to-kafka.md:349]: “查选上游 TiDB”是什么意思？

File: ticdc/ticdc-sink-to-kafka.md [hunk 13]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+* 如果被同步的表有定义 Primary Key，即为 Primary Key。
+* 如果没有 Primary Key，但是有定义 Not NULL Unique Key，即为 Unique Key。
+
+目前，该功能至此 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，用户需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
+
+配置样例如下所示：
+
+```toml
+[sink.kafka-config.large-message-handle]
+# 默认为空，消息超过大小限制后，同步任务失败。
```
Comment (benmaoer) [inline ticdc/ticdc-sink-to-kafka.md:308]:
```suggestion
# 默认为空，消息超过大小限制后，同步任务失败。 该功能从 v7.3.0 开始支持
```

**Review by benmaoer — APPROVED** [review]
LGTM

File: ticdc/ticdc-sink-to-kafka.md [hunk 14]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+    ],
+    "old": null,
+    "_tidb": {     // TiDB 的扩展字段
+        "commitTs": 163963314122145239,
+        "onlyHandleKey": true
+    }
+}
+```
+
+Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。用户需要查选上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```
Comment (3AceShowHand) [inline ticdc/ticdc-sink-to-kafka.md:349]:
```suggestion
Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。用户需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 15]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+    ],
+    "old": null,
+    "_tidb": {     // TiDB 的扩展字段
+        "commitTs": 163963314122145239,
+        "onlyHandleKey": true
+    }
+}
+```
+
+Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。用户需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```
Comment (3AceShowHand) [inline ticdc/ticdc-sink-to-kafka.md:349]:
```suggestion
Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。你需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 16]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
 ```sql
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.TIKV_REGION_STATUS WHERE DB_NAME="database1" AND TABLE_NAME="table1" AND IS_INDEX=0;
 ```
+
+## 处理超过 Kafka Topic 限制的消息
+
+Kafka Topic 对可以接收的消息大小有限制，由 [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:293]:
```suggestion
Kafka Topic 对可以接收的消息大小有限制，该限制由 [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。当 TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 17]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+### 只发送 Handle-Key
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:295]:
```suggestion
### 只发送 Handle Key
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 18]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+
+从 v7.3.0 开始，TiCDC Kafka sink 支持在消息超过限制的时候只发送 Handle-Key 部分数据。这样可以显著减少消息的大小，避免因为消息大小超过 Kafka Topic 限制而导致 changefeed 发生错误和同步任务失败的情况。Handle-Key 指的是：
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:297]:
```suggestion
从 v7.3.0 开始，TiCDC Kafka sink 支持在消息超过限制的时候只发送 Handle Key 部分数据。这样可以显著减少消息的大小，避免因为消息大小超过 Kafka Topic 限制而导致 changefeed 发生错误和同步任务失败的情况。Handle Key 指的是：
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 19]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+```toml
+[sink.kafka-config.large-message-handle]
+# 默认为空，消息超过大小限制后，同步任务失败，该参数 v7.3.0 引入。
+# 设置为 "handle-key-only"，如果消息超过大小，data 字段内容只发送 handle-key，如果依旧超过大小，同步任务失败。
+large-message-handle-option = "handle-key-only"
+```
+
+### 消费只有 Handle-Key 的消息
+
+只有 Handle-key 数据的消息格式如下：
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:315]:
```suggestion
### 消费只有 Handle Key 的消息

只有 Handle Key 数据的消息格式如下：
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 20]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+    ],
+    "old": null,
+    "_tidb": {     // TiDB 的扩展字段
+        "commitTs": 163963314122145239,
+        "onlyHandleKey": true
+    }
+}
+```
+
+Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果存在该字段，那么必然为 true，表示该消息只包含 Handle-Key 部分数据。你需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:349]:
```suggestion
Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果该字段存在且为 true，表示该消息只包含 Handle Key 部分数据。此时，你需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 21]
```diff
@@ -287,3 +287,67 @@ write-key-threshold = 30000
...
+* 如果被同步的表有定义主键，即为主键。
+* 如果没有主键，但是有定义 Not NULL Unique Key，即为 Unique Key。
+
+目前，该功能支持 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，你需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
+
+配置样例如下所示：
+
+```toml
+[sink.kafka-config.large-message-handle]
+# 默认为空，消息超过大小限制后，同步任务失败，该参数 v7.3.0 引入。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:308]:
```suggestion
# 该参数 v7.3.0 引入
# 默认为空，即消息超过大小限制后，同步任务失败
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 22]
```diff
@@ -287,3 +287,68 @@ write-key-threshold = 30000
...
+
+目前，该功能支持 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，你需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
+
+配置样例如下所示：
+
+```toml
+[sink.kafka-config.large-message-handle]
+# 该参数 v7.3.0 引入
+# 默认为空，即消息超过大小限制后，同步任务失败
+# 设置为 "handle-key-only"，如果消息超过大小，data 字段内容只发送 handle-key，如果依旧超过大小，同步任务失败。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:310]:
```suggestion
# 设置为 "handle-key-only" 时，如果消息超过大小，data 字段内容只发送 handle key；如果依旧超过大小，同步任务失败
```

**Review by ran-huang — APPROVED** [review]

File: ticdc/ticdc-sink-to-kafka.md [hunk 23]
```diff
@@ -287,3 +287,68 @@ write-key-threshold = 30000
 ```sql
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.TIKV_REGION_STATUS WHERE DB_NAME="database1" AND TABLE_NAME="table1" AND IS_INDEX=0;
 ```
+
+## 处理超过 Kafka Topic 限制的消息
+
+Kafka Topic 对可以接收的消息大小有限制，该限制由 [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。当 TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 提供了如下解决方案。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:293]:
```suggestion
Kafka Topic 对可以接收的消息大小有限制，该限制由 [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。当 TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 新增一个参数 `large-message-handle-option` 并提供如下解决方案。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 24]
```diff
@@ -287,3 +287,68 @@ write-key-threshold = 30000
...
+
+### 只发送 Handle Key
+
+从 v7.3.0 开始，TiCDC Kafka sink 支持在消息超过限制的时候只发送 Handle Key 部分数据。这样可以显著减少消息的大小，避免因为消息大小超过 Kafka Topic 限制而导致 changefeed 发生错误和同步任务失败的情况。Handle Key 指的是：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:297]:
```suggestion
从 v7.3.0 开始，TiCDC Kafka sink 支持在消息大小超过限制时只发送 Handle Key 的数据。这样可以显著减少消息的大小，避免因为消息大小超过 Kafka Topic 限制而导致 changefeed 发生错误和同步任务失败的情况。Handle Key 指的是：
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 25]
```diff
@@ -287,3 +287,68 @@ write-key-threshold = 30000
...
+
+* 如果被同步的表有定义主键，即为主键。
+* 如果没有主键，但是有定义 Not NULL Unique Key，即为 Unique Key。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:300]:
```suggestion
* 如果被同步的表有定义主键，Handle Key 即为主键。
* 如果没有主键，但是有定义 Not NULL Unique Key，Handle Key 即为 Unique Key。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 26]
```diff
@@ -287,3 +287,68 @@ write-key-threshold = 30000
...
+
+目前，该功能支持 Canal-JSON 和 Open-Protocol 两种编码协议。使用 Canal-JSON 协议时，你需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:302]:
```suggestion
目前，该功能支持 Canal-JSON 和 Open Protocol 两种编码协议。使用 Canal-JSON 协议时，你需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 27]
```diff
@@ -287,3 +287,68 @@ write-key-threshold = 30000
...
+    ],
+    "old": null,
+    "_tidb": {     // TiDB 的扩展字段
+        "commitTs": 163963314122145239,
+        "onlyHandleKey": true
+    }
+}
+```
+
+Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段，如果该字段存在且为 true，表示该消息只包含 Handle Key 部分数据。此时，你需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:350]:
```suggestion
Kafka 消费者收到消息之后，首先检查 `onlyHandleKey` 字段。如果该字段存在且为 `true`，表示该消息只包含 Handle Key 的数据。此时，你需要查询上游 TiDB，通过 [`tidb_snapshot` 读取历史数据](/read-historical-data.md)来获取完整的数据。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 28]
```diff
@@ -287,3 +287,70 @@ write-key-threshold = 30000
...
+* 如果被同步的表有定义主键，主键即为 Handle Key 。
+* 如果没有主键，但是有定义 Not NULL Unique Key，Unique Key 即为 Handle Key。
+
+目前，该功能支持 Canal-JSON 和 Open Protocol 两种编码协议。使用 Canal-JSON 协议时，你需要在 `sink-uri` 中指定 `enable-tidb-extension=true` 参数。
+
+配置样例如下所示：
+
+```toml
+[sink.kafka-config.large-message-handle]
+# 该参数 v7.3.0 引入
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:310]:
```suggestion
# 该参数从 v7.3.0 开始引入
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: ticdc/ticdc-sink-to-kafka.md [hunk 29]
```diff
@@ -287,3 +287,70 @@ write-key-threshold = 30000
...
 SELECT COUNT(*) FROM INFORMATION_SCHEMA.TIKV_REGION_STATUS WHERE DB_NAME="database1" AND TABLE_NAME="table1" AND IS_INDEX=0;
 ```
+
+## 处理超过 Kafka Topic 限制的消息
+
+Kafka Topic 对可以接收的消息大小有限制，该限制由 [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) 参数控制。当 TiCDC Kafka sink 在发送数据时，如果发现数据大小超过了该限制，会导致 changefeed 报错，无法继续同步数据。为了解决这个问题，TiCDC 新增一个参数 `large-message-handle-option` 并提供如下解决方案。
+
+### 只发送 Handle Key
+
+从 v7.3.0 开始，TiCDC Kafka sink 支持在消息大小超过限制时只发送 Handle Key 的数据。这样可以显著减少消息的大小，避免因为消息大小超过 Kafka Topic 限制而导致 changefeed 发生错误和同步任务失败的情况。
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-kafka.md:297]: 从 v 7.3.0 和 7.1.2 开始

**Pull request merged**